File: javascript/7_kyu/simple_string_reversals.js
// Simple string reversal
// https://www.codewars.com/kata/5a71939d373c2e634200008e/train/javascript

function solve(str) {
  var indices = [];

  for (var i = 0; i < str.length; i++) {
    if (str[i] == ' ') {
      indices.push(i);
    }
  }

  var ans = [];

  for (var j = 0; j < str.length; j++) {
    if ( !(str[j] == ' ') ) {
      ans.push(str[j]);
    }
  }

  ans.reverse();

  for (var k = 0; k < indices.length; k++) {
    ans.splice(indices[k], 0, ' ');
  }

  return ans.join('');
}

console.log(solve("your code rocks"));

File: python/7_kyu/simple_string_reversals.py
# Simple string reversal
# https://www.codewars.com/kata/5a71939d373c2e634200008e/train/python

def solve(s):
    indexes = []

    for i in range(0, len(s)):
        if s[i] == ' ':
            indexes.append(i)
    
    ans = [i for i in s if i != ' ']
    ans.reverse()

    for i in indexes:
        ans.insert(i, ' ')
    
    return ''.join(ans)

print(solve("your code rocks"))